File: dataset/human/cpp/file_8930.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
  int n,m;
  cin >> n >> m;
  
  vector<vector<int>> a(n), b(m);
  
  for ( int i = 0; i < n; i++ ){
    string s;
    cin >> s;
    for ( int j = 0; j < n; j++ )
      a[i].push_back(s[j] == '#' ? 1 : 0);
  }
  
  for ( int i = 0; i < m; i++ ){
    string s;
    cin >> s;
    for ( int j = 0; j < m; j++ )
      b[i].push_back(s[j] == '#' ? 1 : 0);
  }
    
  for ( int i = 0; i < n - m + 1; i++ ){
    for ( int j = 0; j < n - m + 1; j++ ){
      bool ok = true;
      for ( int k = 0; k < m; k++ ){
        for ( int l = 0; l < m; l++){
          if ( a[i+k][j+l] == b[k][l] ) continue;
          else{
            ok = false;
            break;
          }
        }
        if ( !ok ) break;
      }
      
      if ( ok ) {
        cout << "Yes" << endl;
        return 0;
      }
    }
  }
  cout << "No" << endl;
  return 0;
}

File: dataset/human/python/file_8930.py
import std.stdio;
import std.string;
import core.stdc.stdlib;

enum vocab = ["dream", "dreamer", "erase", "eraser"];

void solve(string s) {
    if (s == "") {
        writeln("YES");
        exit(0);
    }

    foreach (v; vocab) {
        if (s.startsWith(v)) {
            solve(s[v.length .. $]);
        }
    }
}

void main() {
    string s;
    readf("%s\n", &s);
    solve(s);
    writeln("NO");
}
